Add info on GitHub permissions, fix #195 (#196)
pyOpenSci/software-peer-review: how-to/editor-in-chief-guide.md

Once the package initial checks are complete, and it is determined that
the package is in scope for pyOpenSci review, the Editor in Chief will assign an
editor to the review issue.

> This may involve finding a new editor

Reviewer: ```suggestion
This may involve finding a new (guest) editor
```

as described in the [onboarding guide](onboarding-guide.md).
If you as Editor in Chief do recruit a new editor,
be sure to complete all the onboarding steps described
[here](onboarding-a-new-editor) so that the new editor
has everything they need to manage the review,

> such as GitHub permissions and access to the relevant channels

Reviewer: @NickleDave this is so good. wasn't github permissions a thing we wanted to clarify? i kind of remember you asking me about this but right now i have so many things in my brain i wanted to ask you... 
<img width="467" alt="Screen Shot 2023-03-14 at 4 43 29 PM" src="https://user-images.githubusercontent.com/7649194/225158455-6b5bb8d8-b220-4a16-9535-b84880274810.png">

the proper step is:

Add the new editor to the editorial team so they have permission to modify the template header for the review issue

My question is what permission do YOU need as EIC to add people - can you add them now? do i need to adjust your permissions to be able to do this?

Reviewer: FWIW it just added you as a maintainer of the editorial team. so the process is

1. the peer review lead needs to ensure the EIC has maintainer status on the editorial-board team
2. The Eic can then add a new editor to the team which will give them permissions to modify the template header of each review issue. 

Something like that? thoughts?

> in the pyOpenSci Slack team.

Reviewer: this is great too - can we be specific?

so they need to be added to `private-editorial-team` channel in slack as well?

> It is the editor that will now begin the process of finding reviewers

Reviewer: ```suggestion
The editor will now begin the process of finding reviewers
```

simpler?

for the package.

> [Check out the editor guide for more information on this process](editors-guide.md)

Reviewer: ```suggestion
[Check out the editor guide for more information on the process that an editor follows](editors-guide.md)
```

or something a bit more specific?

As you select an editor (and guide that editor in finding reviewers),
ensure that there is diversity
in the team supporting package review. [Specifically reviewers and the editor should have a mix of gender-identities whenever possible](reviewer-diversity). pyOpenSci [also supports mentoring new reviewers and editors if needed!](review-mentorship)

> 

Reviewer: thank you for this fix!
